sbnobj/Common/Calibration/TrackCaloSkimmerObj.h

```diff
@@ -1,187 +1,109 @@
+    float sumadc   = std::numeric_limits<float>::signaling_NaN();    //!< "SummedADC" -- sum of ADC values under gaussian fit [ADC]
+    float width    = std::numeric_limits<float>::signaling_NaN();    //!< Width of fitted gaussian hit [ticks]
+    float goodness = std::numeric_limits<float>::signaling_NaN();    //!< Goodness-of-fit of fitted gaussian hit
+    float time     = std::numeric_limits<float>::signaling_NaN();    //!< Peak time of hit [ticks]
+    Vector3D sp;                                                     //!< Space-Point Position of hit [cm]
+    int id = -1;                                                     //!< ID of hit
+    uint16_t channel = uint16_t(-1);                                 //!< Channel number of hit
+    uint16_t wire    = uint16_t(-1);                                 //!< Wire number of hit
+    uint16_t plane   = uint16_t(-1);                                 //!< Plane number of hit
+    uint16_t tpc     = uint16_t(-1);                                 //!< TPC number of hit
+    uint16_t mult    = uint16_t(-1);                                 //!< Multiplicity of hit
+    int16_t start = -1;                                              //!< Start tick of hit [ticks]
+    int16_t end   = -1;                                              //!< End tick of hit [ticks]
+    bool hasSP    = false;                                           //!< Whether the hit has a SpacePoint
 
     HitTruth truth;
-
-    HitInfo():
-      integral(-1),
-      sumadc(-1),
-      width(-1),
-      time(-1),
-      id(-1),
-      channel((uint16_t)-1),
-      wire((uint16_t)-1),
-      plane((uint16_t)-1),
-      tpc((uint16_t)-1),
-      mult((uint16_t)-1),
-      start(-1),
-      end(-1) {}
   };
 
   struct TrackHitInfo {
-    HitInfo h; //!< Hit information by itself
-    float pitch; //!< Pitch of track across wire the hit is on [cm]
-    float dqdx; //!< Initial computed dq/dx of hit [ADC/cm]
-    float rr; //!< Residual range of hit along track [cm]
-    Vector3D tp; //!< Track Trajectory position of hit [cm]
-    Vector3D dir; //!< Direction of track at hit location
-    uint16_t i_snippet; //!< Index of hit into snippet 
-    bool ontraj; //!< Whether the hit is on the track trajectory
-    bool oncalo; //!< Whether the hit is on the track calorimetry
-
-    TrackHitInfo():
-      pitch(-1),
-      dqdx(-1),
-      rr(-1),
-      i_snippet((uint16_t)-1),
-      ontraj(false),
-      oncalo(false) {}
-
+    HitInfo h;                                                         //!< Hit information by itself
+    float pitch = std::numeric_limits<float>::signaling_NaN();         //!< Pitch of track across wire the hit is on [cm]
+    float dqdx  = std::numeric_limits<float>::signaling_NaN();         //!< Initial computed dq/dx of hit [ADC/cm]
+    float rr    = std::numeric_limits<float>::signaling_NaN();         //!< Residual range of hit along track [cm]
```

Comment:
Same **breaking change** here too (but in this case all the quantities _are_ defined positive... except maybe for fluctuations on dQ/dx).